docs/content/contribute/tutorial.md

```diff
@@ -183,6 +164,48 @@ the original porter repository and `origin` to refer to your fork:
     main            7e120aab [upstream/main]   Bump cnab-go
     ```
 
+
+### Install Mage
+
+We are transitioning from Make to [Mage]. Installing mage isn't strictly required,
+you can always run `go run mage.go TARGET` instead of `mage TARGET`. However, having
+the tool saves typing and time!
+
+Mage targets are not case-sensitive, but in our docs we use camel case to make
+it easier to read. Run the following commands from the porter directory to install mage:
+
+```bash
+$ go run mage.go EnsureMage
+$ mage
+
+This is a magefile, and is a "makefile for go". See https://magefile.org/
+
+Targets:
+  <List of available targets>
+```
+
+You know that your $GOPATH/bin is configured correctly if you see a list of mage
+targets.
+
+You can enable tab completion for mage as well, so that you can type 
+`mage t[TAB]` and it will complete it with the name of matching targets.
+
+1. Install bash-completion if it isn't already installed with either `brew install
```

Comment:
Would this be applicable to windows folks? I didn't that far into windows development so I'm not sure.

Reply:
At the moment the tutorial says 

> We are improving our support for building Porter on Windows. In a few weeks, we will have it all working on any Windows shell. For now, if you are on Windows, please use Windows Subsystem for Linux (WSL). The rest of this tutorial assumes that you are inside your WSL distribution when installing prerequisites and executing the commands.

So bash completion is something that we can expect from people using WSL.

Reply:
I have been in the process of making _everything_ about our dev process work for Windows (without WSL) but it is taking time. So when I can, I document how to do things in PowerShell, and eventually we will get there! 😀